server/stycoon/game.go

```diff
@@ -2,20 +2,54 @@ package stycoon
 
 import (
 	"database/sql"
+	"fmt"
+	"strconv"
 	"time"
 )
 
 type Game struct {
-	Tick     CurrentTick
-	db       *sql.DB
-	lastTick time.Time
+	Tick          CurrentTick
+	db            *sql.DB
+	lastTick      time.Time
+	ResourceNames map[string]string
+	ShipClasses   map[string]StaticDataShipClassesValue
 }
 
-func NewGame(db *sql.DB) *Game {
+func NewGame(db *sql.DB) (*Game, error) {
 	game := Game{
 		db:       db,
 		lastTick: time.Now(),
 	}
-	game.getGameTick()
-	return &game
+	game.SetGameTick()
+	err := game.SetShipClasses()
+	if err != nil {
+		return &game, err
+	}
+	err = game.SetResourceNames()
+	if err != nil {
+		return &game, err
+	}
+	return &game, nil
+}
+
+func (game *Game) SetResourceNames() error {
+	names := make(map[string]string)
+	rows, err := game.db.Query("select * from d_resource")
+	if err != nil {
+		return fmt.Errorf("query failed %v", err)
+	}
+	var id int
+	var name string
+	for rows.Next() {
+		err = rows.Scan(&id, &name)
```

Comment:
same issue as with the d_classes. d_resources has not changed but that does not mean that it will not change in the future.